src/GitHubCreateMergePRs/Program.cs

```diff
@@ -74,12 +82,39 @@ private static XDocument GetConfig()
         {
             Console.WriteLine($"##vso[task.logissue type=error]Error creating PR. GH response code: {error.StatusCode}");
             Console.WriteLine($"##vso[task.logissue type=error]{await error.Content.ReadAsStringAsync()}");
+
+            // Github rate limits are much lower for unauthenticated users.  We will definitely hit a rate limit
+            // If we hit it during a dryrun, just bail out.
+            if (isDryRun && string.IsNullOrWhiteSpace(githubToken))
+            {
+                if (TryGetRemainingRateLimit(error.Headers, out var remainingRateLimit) && remainingRateLimit == 0)
+                {
+                    Console.WriteLine($"##vso[task.logissue type=error]Hit GitHub rate limit in dryrun with no auth token.  Bailing out.");
```

Comment:
Do we have a run where this line has been hit?

Reply:
In the PR checks :)
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3169149